aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3227,14 +3227,26 @@ static inferSqueezeGeometry(const Tensor &tensor, std::bitset<dim_bitset_size> d
 namespace {
 // Named type instead of a pair/tuple so that we can be sure to
 // construct the vectors in place and get NRVO.
+template <typename T>
 struct InferUnsqueezeGeometryResult {
-  DimVector sizes;
-  DimVector strides;
-  InferUnsqueezeGeometryResult(IntArrayRef tensor_sizes, IntArrayRef tensor_strides)
+  std::vector<T> sizes;
+  std::vector<T> strides;
```

Comment:
Does this need to be `SymDimVector`?

Reply:
Yeah in the "symint" case it would be SymDimVector, but in the int case it would be DimVector. This struct gets used in a couple of other places too, like unsqueeze_quantized, but I wasn't sure if those should be symint-ified as well, so I just made this a template so that it could take in both ints and symints

Reply:
Ah right yea, i forgot there was a quantized one too.

Reply:
This is a deoptimization of the code. Instead, use SmallVector<T, kDimVectorStaticSize>, which is what DimVector resolved to